source/unix/openssl_aes.c

```diff
@@ -0,0 +1,741 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/cal/symmetric_cipher.h>
+
+#include <openssl/evp.h>
+
+struct openssl_aes_cipher {
+    struct aws_symmetric_cipher cipher_base;
+    EVP_CIPHER_CTX *encryptor_ctx;
+    EVP_CIPHER_CTX *decryptor_ctx;
+    struct aws_byte_buf working_buffer;
+};
+
+static int s_encrypt(
+    struct aws_symmetric_cipher *cipher,
+    const struct aws_byte_cursor *input,
+    struct aws_byte_buf *out) {
+    size_t required_buffer_space = input->len + cipher->block_size;
```

Comment:
worth doing aws_add_size_checked()?

Reply:
there's a check before this function gets called that guarantees it can't be too large.